app/src/main/java/com/nextcloud/talk/controllers/ServerSelectionController.kt

```diff
@@ -0,0 +1,373 @@
+import java.security.cert.CertificateException
+import javax.inject.Inject
+
+@AutoInjector(NextcloudTalkApplication::class)
+class ServerSelectionController : NewBaseController(R.layout.controller_server_selection) {
+    private val binding: ControllerServerSelectionBinding by viewBinding(ControllerServerSelectionBinding::bind)
+
+    override val appBarLayoutType: AppBarLayoutType
+        get() = AppBarLayoutType.EMPTY
+
+    @Inject
+    lateinit var ncApi: NcApi
+
+    @Inject
+    lateinit var userUtils: UserUtils
+
+    private var statusQueryDisposable: Disposable? = null
+
+    fun onCertClick() {
+        if (activity != null) {
+            KeyChain.choosePrivateKeyAlias(activity!!, { alias: String? ->
+                if (alias != null) {
+                    appPreferences!!.temporaryClientCertAlias = alias
+                } else {
+                    appPreferences!!.removeTemporaryClientCertAlias()
+                }
+                setCertTextView()
+            }, arrayOf("RSA", "EC"), null, null, -1, null)
+        }
+    }
+
+    override fun onViewBound(view: View) {
+        super.onViewBound(view)
+        sharedApplication!!.componentApplication.inject(this)
+        if (activity != null) {
+            activity!!.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
+        }
+
+        actionBar?.hide()
+
+        binding.hostUrlInputHelperText.setText(
+            String.format(
+                resources!!.getString(R.string.nc_server_helper_text),
+                resources!!.getString(R.string.nc_server_product_name)
+            )
+        )
+        binding.serverEntryTextInputLayout.setEndIconOnClickListener { checkServerAndProceed() }
+        if (resources!!.getBoolean(R.bool.hide_auth_cert)) {
+            binding.certTextView.visibility = View.GONE
+        }
+        if (resources!!.getBoolean(R.bool.hide_provider) ||
+            TextUtils.isEmpty(resources!!.getString(R.string.nc_providers_url)) &&
+            TextUtils.isEmpty(resources!!.getString(R.string.nc_import_account_type))
+        ) {
+            binding.helperTextView.visibility = View.INVISIBLE
+        } else {
+            if ((TextUtils.isEmpty(resources!!.getString(R.string.nc_import_account_type)) ||
+                    findAccounts(userUtils.users as List<UserEntity>).isEmpty()) &&
+                userUtils.users.size == 0
+            ) {
+                binding.helperTextView.setText(R.string.nc_get_from_provider)
+                binding.helperTextView.setOnClickListener {
+                    val browserIntent = Intent(
+                        Intent.ACTION_VIEW, Uri.parse(
+                            resources!!
+                                .getString(R.string.nc_providers_url)
+                        )
+                    )
+                    startActivity(browserIntent)
+                }
+            } else if (findAccounts(userUtils.users as List<UserEntity>).size > 0) {
```

Comment:
somehow always returns 0

Reply:
so SwitchAccountController is not executed at all atm.

Reply:
Also works fine for me, on my device I get 5 accounts from `userUtils`, filtered to 2 via `findAccounts`, do you have several Nextcloud accounts that also have Nextcloud talk _and aren't already added to Nextcloud Talk?

Reply:
ok, didn't really have a look what findAccounts does ;)
so if i understand that right i have to install NC files and add an account there without to add it to NC android talk app. And account manager should find this?? this fails for me..

Reply:
🤦 Approved to fast.

I will test this scenario also.

Reply:
The account manager will find all account configured on the device afaik and the filtering then throws out all which are already in use I think. Nevertheless, the important part is that you get a list > 0 from `userUtils` and this you can check via evaluate in the debugger :) At I can on AS on windows.